openMINDS_validation/validation.py

```diff
@@ -0,0 +1,178 @@
+import re
+from pathlib import Path, PurePath
+
+from openMINDS_validation.utils import load_json, clone_sources_with_specs
+from openMINDS_validation.models import VocabManager, Versions
+
+
+class SchemaValidator(object):
+    def __init__(self, absolute_path):
+        self.absolute_path = absolute_path
+        self.schema = load_json(absolute_path)
+
+        self.version_file = Versions("./versions.json").versions
+
+    def check_required(self):
+        """
+        Validate required properties against the properties defined in the schema definition.
+        """
+        def _check_required_extends(extends_value, required_property):
+            path_prefix_extends = './sources' if extends_value.startswith("/") else './schemas/'
+            path_extends_schema = path_prefix_extends + extends_value
+            extended_schema = load_json(path_extends_schema)
+            if required_property not in extended_schema['properties']:
+                if '_extends' in extended_schema:
+                    return _check_required_extends(extended_schema['_extends'], required_property)
+                raise SyntaxError(f'Missing required property "{required_property}" in the schema definition.')
+            return
+
+        if 'required' not in self.schema:
+            return
+
+        for required_property in self.schema['required']:
+            if required_property not in self.schema['properties'].keys():
+                if '_extends' in self.schema:
+                    _check_required_extends(self.schema['_extends'], required_property)
+                    continue
+                raise SyntaxError(f'Missing required property "{required_property}" in the schema definition.')
+
+    def check_attype(self):
+        """
+        Validate the format of the _type in the schema definition:
+            - First character of _type is in uppercase.
+        """
+        if '_type' in self.schema:
+            type_schema_name = re.split("[:/]", self.schema['_type'])[-1]
+            if not type_schema_name[0].isupper():
+                raise ValueError(f'First character of _type "{type_schema_name}" should be uppercase.')
+
+    def check_extends(self):
+        """
+        Validate existence of schema for the _extends property.
+        """
+        if '_extends' not in self.schema:
+            return
+
+        path_prefix_extends = './sources' if self.schema['_extends'].startswith("/") else './schemas/'
+        path_extends_schema = path_prefix_extends + self.schema['_extends']
+
+        if path_prefix_extends == './sources':
+            clone_sources_with_specs(self.version_file["latest"]["modules"])
```

Comment:
this is not fully correct. could fix this later maybe add a FIXME/TODO to call correct version comparison